docs/topics/compatibility-guides/compatibility-guide-21.md

```diff
@@ -0,0 +1,186 @@
+
+### Deprecate registerKaptGenerateStubsTask(taskName) function
+
+> **Issue**: [KT-70383](https://youtrack.jetbrains.com/issue/KT-70383)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The `registerKaptGenerateStubsTask(taskName)` function is deprecated.
+> Use the new `registerKaptGenerateStubsTask(compileTask, kaptExtension, explicitApiMode)` function instead.
+> This new version allows `KotlinJvmCompile` tasks to be directly connected to `KaptGenerateStubsTask`, making it possible to apply `KotlinJvmCompilerOptions` directly within the task.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: the `registerKaptGenerateStubsTask(taskName)` function is deprecated
+
+### Deprecate KotlinTopLevelExtension and KotlinTopLevelExtensionConfig interfaces
+
+> **Issue**: [KT-71602](https://youtrack.jetbrains.com/issue/KT-71602)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: behavioral
+>
+> **Short summary**: `KotlinTopLevelExtension` and `KotlinTopLevelExtensionConfig` interfaces are deprecated in favor of a new `KotlinTopLevelExtension` interface.
+> This interface merges `KotlinTopLevelExtensionConfig`, `KotlinTopLevelExtension`, and `KotlinProjectExtension`
+> to provide a clearer API hierarchy, and official access to JVM toolchain and compiler properties.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: the `KotlinTopLevelExtension` and `KotlinTopLevelExtensionConfig` interfaces are deprecated
+
+### Remove kotlin-compiler-embeddable from build runtime dependencies
+
+> **Issue**: [KT-61706](https://youtrack.jetbrains.com/issue/KT-61706)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The `kotlin-compiler-embeddable` dependency is removed from the runtime in Kotlin Gradle Plugin (KGP).
+> Required modules are now included directly in KGP artifacts, with a version limit of 2.0 to support compatibility with Gradle versions below 8.2.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report a warning on using `kotlin-compiler-embeddable`
+> - 2.2.0: raise the warning to an error
+
+### Hide compiler symbols from the Kotlin Gradle Plugin API
+
+> **Issue**: [KT-70251](https://youtrack.jetbrains.com/issue/KT-70251)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: Compiler module symbols bundled within the Kotlin Gradle Plugin (KGP), such as `KotlinCompilerVersion`,
+> are hidden from the public API to prevent unintended access in build scripts.
+>
+> **Deprecation cycle**:
+>
+> - 2.1.0: report a warning on accessing these symbols
+> - 2.2.0: raise the warning to an error
+
+### Support for multiple stability configuration files
+
+> **Issue**: [KT-68345](https://youtrack.jetbrains.com/issue/KT-68345)
+>
+> **Component**: Gradle
+>
+> **Incompatible change type**: source
+>
+> **Short summary**: The `stabilityConfigurationFile` property in the Compose compiler options is deprecated
```

Comment:
```suggestion
> **Short summary**: The `stabilityConfigurationFile` property in the Compose extension is deprecated
```

The Compose compiler plugin options are not changed - it is only change in Compose Gradle plugin extension